db/compaction/compaction.cc

```diff
@@ -655,4 +756,32 @@ int Compaction::GetInputBaseLevel() const {
   return input_vstorage_->base_level();
 }
 
+int Compaction::CalcPrecludeBottommostLevel() const {
+  if (!bottommost_level_) {
```

Comment:
I suggest we change it to the last level for simplicity. It's hard to reason that lots of files from different levels can be precluding and a precluding file can stay in the same level as a bottommost file and could be picked up into the same compaction, etc.

Reply:
Discussed with @siying that only have the feature to the last level (instead of bottommost) to simplify the implementation, metrics, etc. So the feature is renamed to `preclude_last_level` instead of `preclude_bottommost`.
We're also plan to make the tiering feature from `bottommost_temperature` to `last_level_temperature`, which should also easier to understand for the user.